hazelcast/src/main/java/com/hazelcast/internal/util/InvocationUtil.java

```diff
@@ -110,6 +123,42 @@ public static LocalRetryableExecution executeLocallyWithRetry(NodeEngine nodeEng
         return execution;
     }
 
+    /**
+     * Invokes the given operation on all cluster members (excluding this
+     * member), in parallel.
+     * <p>
+     * The operation is retried until the cluster is stable between the start
+     * and the end of the invocations.
+     * <p>
+     * The operations invoked with this method should be idempotent.
+     * <p>
+     * If one of the invocations throw any exception other than the
+     * {@link ClusterTopologyChangedException}, {@link MemberLeftException},
+     * {@link TargetNotMemberException}, or
+     * {@link HazelcastInstanceNotActiveException} the method fails with that
+     * exception. When invocations fail with <b>only</b>
+     * {@code ClusterTopologyChangedException}, the invocations are retried.
+     * When invocations fail with <b>only<b/> {@code MemberLeftException},
+     * {@code TargetNotMemberException}, or
+     * {@code HazelcastInstanceNotActiveException} the exceptions are ignored
+     * and the method returns the current member UUIDs, in a similar manner to
+     * {@link #invokeOnStableClusterSerial(NodeEngine, Supplier, int)}.
+     * <p>
+     * Between each retry, the parallel invocations are delayed for
+     * {@link ClusterProperty#INVOCATION_RETRY_PAUSE} milliseconds.
+     *
+     * @return the collection of the member UUIDs that the operations are
+     * invoked on
+     */
+    public static InternalCompletableFuture<Collection<UUID>> invokeOnStableClusterParallel(
+            NodeEngine nodeEngine,
+            Supplier<Operation> operationSupplier,
+            int maxRetries
```

Comment:
I would prefer timeout based version compared to retry count since it is more deterministic for the end user the timeout.

Reply:
I don't want to change it. This is basically the parallel counterpart of the widely used invokeOnStableClusterSerial and it uses count-based retries